lib/rubocop/cop/lint/shadowing_outer_local_variable.rb

```diff
@@ -56,19 +56,26 @@ def before_declaring_variable(variable, variable_table)
 
           outer_local_variable = variable_table.find_variable(variable.name)
           return unless outer_local_variable
+          return if variable_used_in_declaration_of_outer?(variable, outer_local_variable)
           return if same_conditions_node_different_branch?(variable, outer_local_variable)
 
           message = format(MSG, variable: variable.name)
           add_offense(variable.declaration_node, message: message)
         end
 
+        private
+
+        def variable_used_in_declaration_of_outer?(variable, outer_local_variable)
+          variable.scope.node.each_ancestor.any?(outer_local_variable.declaration_node)
+        end
+
         def same_conditions_node_different_branch?(variable, outer_local_variable)
           variable_node = variable_node(variable)
           return false unless node_or_its_ascendant_conditional?(variable_node)
 
           outer_local_variable_node =
             find_conditional_node_from_ascendant(outer_local_variable.declaration_node)
-          return true unless outer_local_variable_node
+          return false unless outer_local_variable_node
```

Comment:
This used to return `true` only for this example: https://github.com/rubocop/rubocop/blob/2e6253c9e8db8784be31f79939d4625aea7e7803/spec/rubocop/cop/lint/shadowing_outer_local_variable_spec.rb#L313-L323

However, I don't think `true` should be returned here because this method is not really the place for this check (at least according to the method's name). That's why I changed it to `false` here and extracted the above example's logic to `variable_used_in_declaration_of_outer?` method which I think is more correct.